ipam/ipam.go

```diff
@@ -129,21 +149,30 @@ func (ipam *IPAM) legacyAllocateIpByName(input interface{}, ctx common.RestConte
 		return nil, err
 	}
 	found = false
-	var i int
-	for i = range tenants {
-		if tenants[i].Name == tenantName {
-			found = true
-			Endpoint.TenantId = fmt.Sprintf("%d", tenants[i].ID)
-			log.Printf("IPAM: Tenant name %s has ID %s, original %d\n", tenantName, Endpoint.TenantId, tenants[i].ID)
+	for _, t := range tenants {
+		switch tenantLookupField {
+		case "Name":
+			if t.Name == tenantParam {
```

Comment:
There's reflection, but this isn't a good use-case for it.
`reflect.ValueOf(t).FieldByName(field).String()`
You'd use that kind of thing if you had arbitrary types involved, or an appropriate amount of complexity to deal with.